eZ/Publish/LegacySearch/RecommendationLegacySearchEngine.php

```diff
@@ -80,4 +80,60 @@ public function addNodeAssignment($mainNodeID, $objectID, $nodeAssignmentIDList,
             $this->legacySearchEngine->addNodeAssignment($mainNodeID, $objectID, $nodeAssignmentIDList, $isMoved);
         }
     }
+
+    public function cleanup()
+    {
+        if (method_exists($this->legacySearchEngine, 'cleanup')) {
+            return $this->legacySearchEngine->cleanup();
+        }
+    }
+
+    public function updateObjectsSection($objectIDs, $sectionID)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateObjectsSection')) {
+            return $this->legacySearchEngine->updateObjectsSection($objectIDs, $sectionID);
+        }
+    }
+
+    public function updateObjectState($objectID, $objectStateList)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateObjectState')) {
+            return $this->legacySearchEngine->updateObjectState($objectID, $objectStateList);
+        }
+    }
+
+    public function updateNodeSection($nodeID, $sectionID)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateNodeSection')) {
+            return $this->legacySearchEngine->updateNodeSection($nodeID, $sectionID);
+        }
+    }
+
+    public function updateNodeVisibility($nodeID, $action)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateNodeVisibility')) {
```

Comment:
Perhaps so, however:
- There may be other missing functionality that I'm not fully aware of, exactly which events/data should be sent updated in recommendationClient ?
- I believe it would be a separate issue, as this refers specifically to the real legacy search engine not indexing properly on certain events.

IMHO a follow-up ticket should be created for reco-specific indexing.